### DISCUSSION TITLE
How to PATCH users using the API?

### DISCUSSION THREAD

**Discussion by ferrlen** [kimai/kimai#3115] [Questions & Answers]
Reading the doc here https://demo.kimai.org/api/doc it says that I can update the info on any users via the API with a PATCH request, including in the body of the request a JSON object with the attributes I want to change.

For example:

I want to update the information on user `https://demo.kimai.org/api/users/42` ; specifically, I want to change `"enabled": true` and add the user to a specific team.

I'd thus set as body for the PATCH request:

```
{
  "enabled": true,
  "teams": [
        {
            "id": 3,
            "name": "My Team",
            "color": null
        }
    ]
}
```

(do I also need to update the `"membership"` attribute, or is updating `"teams"` enough?)

When I try to submit the previous request, I get a `200OK` response that changes nothing, it gives me the same JSON object the user had previous to my PATCH request. Retrieving the information on the user again via the API or by logging in with the browser confirms that nothing was changed (i.e., user still has `"enabled": false` and `"teams": []`). 

What am I missing? Why isn't my PATCH request changing anything?

**Comment by kevinpapst** [comment]
Where do you see `teams` in this request?
<img width="729" alt="Bildschirmfoto 2022-02-01 um 02 23 59" src="https://user-images.githubusercontent.com/533162/151899542-860f3fad-caf1-4093-a98b-24cf49246663.png">

Anyways, sending a PATCH with the payload `{ "enabled": true}` enables the user.

Adding a user to a team requires the use of the teams endpoint: `/api/teams/{id}/members/{userId}`

**Reply by ferrlen** [reply]
> Where do you see `teams` in this request?
> 
> Anyways, sending a PATCH with the payload `{ "enabled": true}` enables the user.
> 
> Adding a user to a team requires the use of the teams endpoint: `/api/teams/{id}/members/{userId}`

Thanks for the fast reply! I didn't know whether that list was exhaustive or if it was possible to include `teams` too etc. I had tried already just `{"enabled": true}` and `{"alias": "something"}`, but neither work. I tried then adding a member via `/api/teams/{id}/members/{userId}` and it worked perfectly.

I wonder if I lack a permission? I can change the `enabled` value on the website as an `Administrator`. On the other hand, I get a `200OK` response for my PATCH request, but spitting back the user info exactly as it was before my request, nothing is changed.

**Reply by kevinpapst** [reply]
You can test yourself if this is a permission issue by using a ROLE_SUPER_ADMIN account. If it works then, you can check the list of permissions in the documentation and compare which one is missing for admins.
